stubs/python-dateutil/dateutil/tz/tz.pyi

```diff
@@ -84,12 +93,21 @@ class tzrange(tzrangebase):
 
 class tzstr(tzrange):
     hasdst: bool
-    def __init__(self, s: bytes | _FileObj, posix_offset: bool = ...) -> None: ...
+    def __init__(self, s: str, posix_offset: bool = ...) -> None: ...
     @classmethod
     def instance(cls, name, offset) -> tzoffset: ...
 
+class _ICalReader(Protocol):
+    # optional attribute:
+    # read: str
+    def __enter__(self: Self) -> Self: ...
+    def __exit__(
+        self, exc_type: type[BaseException] | None, exc_value: BaseException | None, traceback: TracebackType | None
+    ) -> bool | None: ...
```

Comment:
These aren't needed, because the file object is wrapped in `_nullcontext`. It doesn't close files passed in, it only closes the file if a string was passed in and it opened the file.